content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb

```diff
@@ -120,8 +146,35 @@ def vmdb_object(model, id)
 end
 
 def requested_storage(args_hash)
-  vm_size = args_hash[:resource].vm_template.provisioned_storage
-  args_hash[:prov_value] = args_hash[:number_of_vms] * vm_size
+  if @reconfigure_request
+    args_hash[:prov_value] = 0
+    # Adding/removing disks only supported for VMware
+    if args_hash[:resource].options[:disk_add]
+      args_hash[:resource].options[:disk_add].each do |disk|
+        $evm.log(:info, "Adding a disk:  #{disk.inspect}")
+        args_hash[:prov_value] += disk['disk_size_in_mb'].to_i.megabytes
+      end
+    end
+    if args_hash[:resource].options[:disk_remove]
+      args_hash[:resource].options[:disk_remove].each do |disk|
+        $evm.log(:info, "Removing disk: #{disk.inspect}")
+        disk_num = disk[:disk_name].match(/_(\d).vmdk/)
+        next unless disk_num
+        disk_n_number = "disk_#{disk_num[1].succ}_size"
+        disk_n_size   = @vm.send(disk_n_number.to_s)
+        $evm.log(:info, "Removing disk size: #{disk_n_size.to_s(:human_size)}") if disk_n_size
```

Comment:
@billfitzgerald0120 
can use use next unless disk_n_size and change the log message to 
$evm.log(:info, "Disk size:  #{disk_n_size.to_s(:human_size)}")